skrub/_table_vectorizer.py

```diff
@@ -671,10 +573,18 @@ def _check_X(self, X):
             # Create a copy to avoid altering the original data.
             X = X.copy()
 
+        # Check for duplicate column names.
+        duplicate_columns = {k for k, v in Counter(X.columns).items() if v > 1}
+        if len(duplicate_columns) > 0:
```

Comment:
I can't comment on it but in this function it seems we copy X if it is a dataframe but not if it is an array.
but we will remove support for arrays anyway in #831.
also we already copy all columns in replace_missing_indicators so we should avoid the `X.copy()`; we can deal with that in one of the other PRs